components/BottomNavigation/src/private/MDCBottomNavigationCellBadge.m

```diff
@@ -0,0 +1,120 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCBottomNavigationCellBadge.h"
+#import "MaterialMath.h"
+
+static const CGFloat kMDCBottomNavigationCellBadgeAngle = 90.f;
+static const CGFloat kMDCBottomNavigationCellBadgeFontSize = 10.f;
+static const CGFloat kMDCBottomNavigationCellBadgeXPadding = 8.f;
+static const CGFloat kMDCBottomNavigationCellBadgeYPadding = 2.f;
+
+@implementation MDCBottomNavigationCellBadge
+
+- (instancetype)init {
```

Comment:
In general, we typically only implement designated initializers.
So in this case, UIView, it would be initWithFrame: and initWithCoder: